src/main/java/io/appform/dropwizard/sharding/dao/LockedContext.java

```diff
@@ -142,11 +139,26 @@ public <U> LockedContext<T> update(RelationalDao<U> relationalDao, Object id, Fu
     public <U> LockedContext<T> createOrUpdate(
             RelationalDao<U> relationalDao,
             DetachedCriteria criteria,
-            Function<U, U> updater,
+            UnaryOperator<U> updater,
```

Comment:
This will be a breaking change for some people

Reply:
It's interface compatible so it should be fine

Reply:
`public interface UnaryOperator<T> extends Function<T, T>`